src/website/app/Router.js

```diff
@@ -129,7 +134,15 @@ export default function Router({ demoRoutes }: Props) {
       <Route
         path="/components/:componentId"
         render={({ location, match }) => {
-          const componentId = match.params.componentId || firstDemoSlug;
+          const componentId =
+            flatDemoRoutes.find(
+              route => route.slug === match.params.componentId
+            ) && match.params.componentId;
+
+          if (!componentId) {
+            return <Redirect to={`/components/${firstDemoSlug}`} />;
```

Comment:
I am assuming that you weighed the benefit an incorrect component route going to the 404 page vs. it going to the first component and decided that the latter was better.  Is that correct?

Reply:
Yep. Seemed like an opinionated redirect was more helpful.